src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -130,29 +135,32 @@ public static unsafe ulong BigMul(ulong a, ulong b, out ulong low)
 
             static ulong SoftwareFallback(ulong a, ulong b, out ulong low)
             {
-                // It's adoption of algorithm for multiplication
+                // Adaptation of algorithm for multiplication
                 // of 32-bit unsigned integers described
                 // in Hacker's Delight by Henry S. Warren, Jr. (ISBN 0-201-91465-4), Chapter 8
                 // Basically, it's an optimized version of FOIL method applied to
                 // low and high dwords of each operand
-                const ulong lowBitsMask = 0xFFFFFFFFU;
 
-                ulong al = a & lowBitsMask;
+                ulong al = (uint)a;
                 ulong ah = a >> 32;
```

Comment:
Do you think it worth fixing in RyuJIT?

Reply:
Doesn't regular `MUL` return the 128 bit result across the two registers `RDX:RAX` so it could be an intrinsic also?

Reply:
Though MULX is probably better; so don't know :)

Reply:
> Do you think it worth fixing in RyuJIT?

It is one of those pattern optimizations that help a tiny bit. We have a lot of them already, so one more would not hurt.

>  it could be an intrinsic also?

Yes, it could be intrinsic. We have a problem with modeling intrinsics that return two bits of information in the JIT. DivMod has the same problem.

> MULX is probably better

JIT can expand the intrinsic into MULX as appropriate...